### ISSUE TITLE
Integrating complete Wikipedia Articles in POI discriptions is a bad decision

### ISSUE THREAD

**Issue by shorty66** [osmandapp/Osmand#1925]
The POIs should rather link to APPs specifically designed to view wikipedia articles.
These Apps include Aard2, Kiwix and the official Wikipedia App. If no such App is installed the intent could be handled by the browser.

Removing article-content form the POI files should result in a much smaller file size.

**Comment by jkufner** [comment]
I see three options:

  1. Whole article in POI description, so Osmand can handle everything. Wiki POI is POI just like any other. Current implementation.
  2. POI is only a link to Wikipedia application. Osmand has list of POI (coordinates and ID), but does not have content. Much smaller size, but requires Wiki app.
  3. Osmand asks Wikipedia app whether there are POIs on current map view. This will require modification in Wikipedia app, but user will have wiki POIs only once on his device. This will also open a way to implement additional external providers, like calendar application which would export coordinates of events and show them on map.

**Comment by vshcherb** [comment]
1 and 2 are already implemented. If you contribute to OSM, you can put wiki link and it will be nicely shown without downloading separate POI Wikipedia and that intent will be caught by WIkipedia offline ap.

**Comment by shorty66** [comment]
I did not know about wiki links in regular POIs.
It would be nice to have an option to show all regular POIs containing wiki links, though.

**Comment by DFyson** [comment]
I think both internal and external wikipedia article support is important.
The wikipedia URLs can be picked up by external apps so long as they have an [intent](https://en.wikipedia.org/wiki/Intent_%28Android%29) filter setup in its manifest which matches wikipedia URLs. I have Aard2, Kiwix and Wikipedia app. Of those, only the Wikipedia app works with the links along side with the web browsers. The code responsible for that is in [its manifest file](https://github.com/wikimedia/apps-android-wikipedia/blob/master/app/src/main/AndroidManifest.xml). So Kiwix needs to add support for it and Aard2 seems to have [something](https://github.com/itkach/aard2-android/commit/05cdde1d0fd1dc516000dbd5adc287b4c57022db) but it's clearly not working.

There are also two types of POIs with wikipedia references here. 1: Objects in the map file which contain the wikipedia tag linking to a wikipedia article. 2: POIs from the wikipedia file (which has to be downloaded independently to the map file) which is a map of all the geo-tagged wikipedia pages and contains the full articles.
